python/lsst/ip/isr/brighterFatterKernel.py

```diff
@@ -0,0 +1,413 @@
+# This file is part of ip_isr.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+"""Brighter Fatter Kernel calibration definition."""
+
+
+__all__ = ['BrighterFatterKernel']
+
+
+import numpy as np
+from astropy.table import Table
+import lsst.afw.math as afwMath
+from . import IsrCalib
+
+
+class BrighterFatterKernel(IsrCalib):
+    """Calibration of brighter-fatter kernels for an instrument.
+
+    ampKernels are the kernels for each amplifier in a detector, as
+    generated by having level == 'AMP'
+
+    detectorKernel is the kernel generated for a detector as a
+    whole, as generated by having level == 'DETECTOR'
+
+    makeDetectorKernelFromAmpwiseKernels is a method to generate the
+    kernel for a detector, constructed by averaging together the
+    ampwise kernels in the detector.
```

Comment:
`makeDetectorKernelFromAmpwiseKernels`: in terms of usage, do we get this when selecting `AMP`, or when selecting `DETECTOR`, or in both cases? Perhaps it would be good to clarify in the docstring.

Reply:
Upon reading further (`isrTask.py`), I see that there's two options to control this: `brighterFatterKernel.level` and `self.config.brighterFatterLevel`., and that `makeDetectorKernelFromAmpwiseKernels` happens when `self.config.brighterFatterLevel == 'DETECTOR'` and `brighterFatterKernel.level == AMP`. So maybe you can add this info here too?

Reply:
I've updated the docstring.  At some point, there will probably be questions about why the application code expects {{level == 'DETECTOR'}}, but the generation code defaults to {{level == 'AMP'}}.